bimmer_connected/vehicle.py

```diff
@@ -185,9 +185,9 @@ def available_attributes(self) -> List[str]:
             result += LIDS
             result += WINDOWS
             result += self.drive_train_attributes
-            result += ['DCS_CCH_Activation', 'DCS_CCH_Ongoing', 'condition_based_services', 'check_control_messages',
-                       'door_lock_state', 'internalDataTimeUTC', 'lights_parking', 'lids', 'windows',
-                       'positionLight', 'last_update_reason', 'singleImmediateCharging']
+            result += ['DCS_CCH_Activation', 'DCS_CCH_Ongoing', 'condition_based_services',
+                       'check_control_messages', 'door_lock_state', 'internalDataTimeUTC',
+                       'parking_lights', 'positionLight', 'last_update_reason', 'singleImmediateCharging']
```

Comment:
Well, because the test says so 😄 
- `lights_parking`to `parking_lights`: I only found the latter in `state.py`, i.e. [here]( https://github.com/bimmerconnected/bimmer_connected/blob/ef7b25e412a609b23a2d2f29eb4e23558554bc25/bimmer_connected/state.py#L294). This fixes one testcase, as `lights_parking` is not an attribute of `vehicle.state`.
- `windows` and `lids` is not found in any other code or `status.json` at all. 

Maybe you did try something out and added it by chance? I've search all vehicle fingerprints but never found these attributes.

Reply:
Let me check that, because I think I then use that in the HA integration, otherwise it doesn't make any sense indeed.

Reply:
I do recall now (at least I think so :wink:).
I have added some logic in HA last year to only add (binary) sensors when these actually exist in bimmer_connected -> https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/bmw_connected_drive/binary_sensor.py#L13-L20. This way users don't get any sensors for which no data is available.

And `lids`, `windows` and `lights_parking` where already existing binary sensors (and these names are used in the entity) for which I didn't want to make a breaking change. That's why I added these in there. Maybe not the cleanest way (just an amateur over here), but removing them will result in HA missing 3 binary sensors, so that's not good.
Is there an easy way you know of to make the test ignore these?

Reply:
Ah that makes sense :)
The magic lies in `test/__init__.py`'s `ADDITIONAL_ATTRIBUTES` and `MISSING_ATTRIBUTES`. Whoever built this originally (m1n3rva?) did some great work!

Reply:
Yeah it was initially build by someone else and later m1n3rva took over. They both did a really great job.

The first version of the library was using a different API which used amongst others `lights_parking`. We switched that API shortly after the integration was added in HA, but wanted to prevent breaking changes as much as possible.